driver.go

```diff
@@ -48,13 +48,33 @@ func (l *lvmDriver) Create(req volume.Request) volume.Response {
 		return resp(err)
 	}
 
-	cmdArgs := []string{"-n", req.Name}
+	cmdArgs := []string{"-n", req.Name, "--setactivationskip", "n"}
+	snap, ok := req.Options["snapshot"]
+	isSnapshot := ok && snap != ""
 	s, ok := req.Options["size"]
-	if !ok || (ok && s == "") {
+	hasSize := ok && s != ""
+
+	// no size is valid for thinly-provisioned snapshot volumes
+	if !hasSize && !isSnapshot {
```

Comment:
@cschauer I have added a helper function `isThinlyProvisioned()` which will return you a bool (true/false) indicating if your volume is a regular or thin volume. You can use that to solve this issue. 

"Docker 1.12 adds a Status field in the response to VolumeDriver.get": Let's open a separate PR for this.

Shishir